### DISCUSSION TITLE
Allow configuration of `list` attribute for `<input type="color">`

### DISCUSSION THREAD

**Discussion by mrleblanc101** [laravel/nova-issues#4720] [Ideas & Feature Requests] [answered]
HTML 5 allow to add predefined color for `<input type="color">` with the `list` attribute.
Currently, we can't define the colors with `Color::make()`.
It would be cool if we could.
<img width="234" alt="Capture d’écran, le 2022-08-19 à 13 24 40" src="https://user-images.githubusercontent.com/17392251/185674228-b9aea253-7881-40ae-9cb0-873079c09d57.png">

**Comment by mrleblanc101** [comment]
Also noticed the color input gets a `cursor: not-allowed`.
It match `:read-only` even if there is no `read-only` attribute.
![Capture d’écran, le 2022-08-19 à 15 25 09](https://user-images.githubusercontent.com/17392251/185694752-2816ddad-e8ea-4005-b570-a0865f58f3f6.png)

**Reply by crynobone** [reply]
It seems browser by default define `<input type="color">` as read-only, we don't have enough information to verify this but Nova doesn't have anything that enforce it to behave as read-only.

**Reply by mrleblanc101** [reply]
Thank you, I thought it might be a browser issue. Any thought on the color presets customization ?

**Reply by crynobone** [reply]
Submitted a PR to support `Color::make()->suggestions()` but be aware that not all browsers support this option. For example, it doesn't do anything on Firefox.

PR is pending review at the moment.

**Reply by mrleblanc101** [reply]
The funny thing is that the `list` attribute should reference a `<datalist>` of the ID from the list attribute value, in this case `color-list`. 

But there is not datalist and nothing with the id `color-list` in the Nova frontend. 

That's why I noticed, because removing the attribute would open the color picker directly ! I wasn't even aware of this attribute until now.

I guess that Chrome has a default list set if the attribute value is invalid.

**Comment by crynobone** [comment, marked as answer]
You can now use `Color::make()->suggestions(['#FFFFFF', '#000000'])` etc starting from `4.13.0`.
